bin/functions/load-config.py

```diff
@@ -327,152 +307,231 @@ def generate_optional_value():  # get some critical values from environment or m
-        if HibenchConf['hibench.hadoop.release'] == 'apache' and HibenchConf["hibench.hadoop.version"] == "hadoop1":
+        if HibenchConf['hibench.hadoop.release'] == 'apache':
             HibenchConf["hibench.sleep.job.jar"] = HibenchConf['hibench.hadoop.examples.jar']
-            HibenchConfRef["hibench.sleep.job.jar"] = "Refer to `hibench.hadoop.examples.jar` according to the evidence of `hibench.hadoop.release` and `hibench.hadoop.version`"
+            HibenchConfRef[
+                "hibench.sleep.job.jar"] = "Refer to `hibench.hadoop.examples.jar` according to the evidence of `hibench.hadoop.release`"
         else:
             log("probe sleep jar:", HibenchConf['hibench.hadoop.examples.test.jar'])
             HibenchConf["hibench.sleep.job.jar"] = HibenchConf['hibench.hadoop.examples.test.jar']
-            HibenchConfRef["hibench.sleep.job.jar"] = "Refer to `hibench.hadoop.examples.test.jar` according to the evidence of `hibench.hadoop.release` and `hibench.hadoop.version`"
+            HibenchConfRef[
+                "hibench.sleep.job.jar"] = "Refer to `hibench.hadoop.examples.test.jar` according to the evidence of `hibench.hadoop.release`"
 
+def probe_hadoop_configure_dir():
     # probe hadoop configuration files
     if not HibenchConf.get("hibench.hadoop.configure.dir", ""):
-        if HibenchConf["hibench.hadoop.release"] == "apache" or HibenchConf["hibench.hadoop.release"] == "hdp": # Apache and HDP release
-            HibenchConf["hibench.hadoop.configure.dir"] = join(HibenchConf["hibench.hadoop.home"], "conf") if HibenchConf["hibench.hadoop.version"] == "hadoop1" \
-                else join(HibenchConf["hibench.hadoop.home"], "etc", "hadoop")
-            HibenchConfRef["hibench.hadoop.configure.dir"] = "Inferred by: 'hibench.hadoop.version' & 'hibench.hadoop.release'"
-        elif HibenchConf["hibench.hadoop.release"].startswith("cdh"): # CDH release
-            HibenchConf["hibench.hadoop.configure.dir"] = join(HibenchConf["hibench.hadoop.home"], "etc", "hadoop-mapreduce1") if HibenchConf["hibench.hadoop.version"] == "hadoop1" \
-                else join(HibenchConf["hibench.hadoop.home"], "etc", "hadoop")
-            HibenchConfRef["hibench.hadoop.configure.dir"] = "Inferred by: 'hibench.hadoop.version' & 'hibench.hadoop.release'"
-
+        # Apache and HDP release
+        if HibenchConf["hibench.hadoop.release"] == "apache" or HibenchConf["hibench.hadoop.release"] == "hdp":
+            HibenchConf["hibench.hadoop.configure.dir"] = join(HibenchConf["hibench.hadoop.home"], "etc", "hadoop")
+            HibenchConfRef["hibench.hadoop.configure.dir"] = "Inferred by: 'hibench.hadoop.release'"
+        # CDH release
+        elif HibenchConf["hibench.hadoop.release"].startswith("cdh"):
+            HibenchConf["hibench.hadoop.configure.dir"] = join(HibenchConf["hibench.hadoop.home"], "etc", "hadoop")
+            HibenchConfRef["hibench.hadoop.configure.dir"] = "Inferred by:  & 'hibench.hadoop.release'"
```

Comment:
There is no difference for apache, hdp, cdh anymore? Can we combine these and removing the if else.